### PR TITLE
refactor: update shutdown process

### PR DESCRIPTION
**Pull request by mkurapov**
## Changes proposed in this pull request

- Moves ILP connector server into app, instead of having it separately defined in `index.ts`
- Updates `app.shutdown` function to properly close all running servers (in both `auth` and `backend` packages)
- Make sure we are correctly handling multiple SIGTERM events
- Remove unused `closeEmitter`

## Context

Related to: https://github.com/interledger/rafiki/pull/1805#discussion_r1313026526

## Checklist

- [ ] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [x] Make sure that all checks pass
- [ ] Postman collection updated

### CODE REVIEW

File: packages/backend/src/app.ts [hunk 1]
```diff
@@ -207,21 +206,21 @@ export interface AppServices {
   feeService: Promise<FeeService>
   autoPeeringService: Promise<AutoPeeringService>
   autoPeeringRoutes: Promise<AutoPeeringRoutes>
+  connectorApp: Promise<ConnectorApp>
+  tigerbeetle: Promise<TigerbeetleClient>
```
Comment (mkurapov) [inline packages/backend/src/app.ts:210]: adds these services so they are properly typed when doing `container.use`

File: packages/backend/src/app.ts [hunk 2]
```diff
@@ -74,10 +74,10 @@ import { CombinedPaymentService } from './open_payments/payment/combined/service
 import { FeeService } from './fee/service'
 import { AutoPeeringService } from './auto-peering/service'
 import { AutoPeeringRoutes } from './auto-peering/routes'
+import { Rafiki as ConnectorApp } from './connector/core'
 
 export interface AppContextData {
   logger: Logger
-  closeEmitter: EventEmitter
```
Comment (mkurapov) [inline packages/backend/src/app.ts:80]: wasn't used

File: packages/backend/src/index.ts [hunk 3]
```diff
@@ -445,19 +441,12 @@ export const gracefulShutdown = async (
...
-      )
-    })
-  }
   const knex = await container.use('knex')
   await knex.destroy()
   const tigerbeetle = await container.use('tigerbeetle')
-  await tigerbeetle.destroy()
+  tigerbeetle.destroy()
   const redis = await container.use('redis')
-  await redis.disconnect()
```
Comment (mkurapov) [inline packages/backend/src/index.ts:460]: not async functions

File: packages/backend/src/shared/ilp_plugin.ts [hunk 4]
```diff
@@ -37,6 +37,6 @@ export class OutgoingIlpPlugin implements IlpPlugin {
 }
 
 export interface IlpPluginOptions {
-  sourceAccount: LiquidityAccount
+  sourceAccount: ConnectorAccount
```
Comment (mkurapov) [inline packages/backend/src/shared/ilp_plugin.ts:40]: updating to the the correct type

File: packages/auth/src/app.ts [hunk 5]
```diff
@@ -158,12 +152,12 @@ export class App {
     })
 
     // Setup Apollo
-    this.apolloServer = new ApolloServer({
+    const apolloServer = new ApolloServer({
```
Comment (mkurapov) [inline packages/auth/src/app.ts:155]: this.apolloServer wasn't used outside of this function

**Review by sabineschaller — APPROVED** [review]
Nice cleanup! It'll probably give me merge conflicts 🙃

**Review by raducristianpopa — APPROVED** [review]

**Pull request merged**